### PR TITLE
Fix the compute_output_shape issue in tf.keras.layers.Bidirectional

### LINKED ISSUE #23299: `tf.keras.layers.Bidirectional` does not work under eager execution mode
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): NO
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS X 10.12.6
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: NO
- TensorFlow installed from (source or binary): BINARY
- TensorFlow version (use command below): v1.12.0-rc0-17-g7b08198113 1.12.0-rc1
- Python version: 3.6.5
- Bazel version (if compiling from source): NA
- GCC/Compiler version (if compiling from source): NA
- CUDA/cuDNN version: NA
- GPU model and memory: NA

**Describe the current behavior**

A very simple model with `Bidirectional` layer will stop work after we enable the eager execution mode.

**Describe the expected behavior**

Code should work with eager execution mode.

**Code to reproduce the issue**

The following code will be able to reprorduce this problem:

```py
import tensorflow as tf
import numpy as np

tf.enable_eager_execution()

x = tf.keras.layers.Input(shape=(2, 1))
y = tf.keras.layers.Bidirectional(
    tf.keras.layers.LSTM(1, return_state=True)
)(x)

model = tf.keras.Model(inputs=x, outputs=y)

data = np.array([0.1, 0.2]).reshape((1, -1, 1))
print(model.predict(data))
```
However, if we comment out the `tf.enable_eager_execution()`, then it will work as expected:

**Other info / logs**

The error message(with eager execution enabled) is:

```
Traceback (most recent call last):
  File "src/t.py", line 11, in <module>
    )(x)
  File "/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/layers/wrappers.py", line 473, in __call__
    return super(Bidirectional, self).__call__(inputs, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/engine/base_layer.py", line 769, in __call__
    output_shapes = self.compute_output_shape(input_shapes)
  File "/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/utils/tf_utils.py", line 149, in wrapper
    output_shape = fn(instance, input_shape)
  File "/usr/local/lib/python3.6/site-packages/tensorflow/python/keras/layers/wrappers.py", line 444, in compute_output_shape
    input_shape).as_list())
AttributeError: 'list' object has no attribute 'as_list'
```

After we disable the eager execution, then the above code works as expected.

```
[array([[0.01535364, 0.04371894]], dtype=float32), array([[0.01535364]], dtype=float32), array([[0.03121366]], dtype=float32), array([[0.04371894]], dtype=float32), array([[0.09061633]], dtype=float32)]
```

### PR DESCRIPTION
**Pull request by feihugis**
This PR fixes #23299, where `tf.keras.layers.Bidirectional` does not work in some cases under eager execution mode because the output shape from the forward layer may not be as expected.

### CODE REVIEW

**Comment by feihugis** [comment]
@alextp Could you please help review this PR when you have time?

**Review by huan — DISMISSED** [review]
LGTM

File: tensorflow/python/keras/layers/wrappers.py [hunk 1]
```diff
@@ -441,10 +441,15 @@ def set_weights(self, weights):
 
   @tf_utils.shape_type_conversion
   def compute_output_shape(self, input_shape):
-    output_shape = tuple(self.forward_layer.compute_output_shape(
-        input_shape).as_list())
+    forward_layer_output_shape \
```
Comment (alextp) [inline tensorflow/python/keras/layers/wrappers.py]: can you avoid using `\` to designate line breaks? This is tensorflow python style (just break after the parenthesis)

**Review by alextp — CHANGES_REQUESTED** [review]

**Comment by feihugis** [comment]
@alextp Thanks for your review. `\` has been removed, and add the break after the parenthesis. Could you review the [change](https://github.com/tensorflow/tensorflow/pull/23441/commits/b11c33a5d1792ba7e4ff0a687d43c0322b3458d0) ?

File: tensorflow/python/keras/layers/wrappers.py (hunk 1 above)
Comment (feihugis) [reply tensorflow/python/keras/layers/wrappers.py]: `\` has been removed and add the break after the parenthesis.

**Review by alextp — APPROVED** [review]

**Comment by feihugis** [comment]
@alextp @fchollet This PR has been `ready to pull` for a while. Is there anything else I need to do for merging?

**Comment by qlzh727** [comment]
This PR is no longer needed. See https://github.com/tensorflow/tensorflow/issues/23299 for more details.

**Pull request closed without merging**